instaloader/structures.py

```diff
@@ -160,7 +160,19 @@ def _optional_normalize(string: Optional[str]) -> Optional[str]:
         return normalize("NFC", string)
     else:
         return None
-
+def is_fake_mp4(url: str) -> bool:
+    """
+    Detect if a video URL is actually a 1-frame JPEG.
+    """
+    try:
+        headers = {"Range": "bytes=0-1023"}  # only first KB
+        r = requests.get(url, headers=headers, timeout=5)
```

Comment:
This introduces a request that is very specific, likely never sent by any browser and would make it trivial to detect Instaloader. This feature must be implemented ideally without a extra request.